### PR TITLE
README: Updates

### PR DESCRIPTION
**Pull request by tgraf**
* Rework components into high level feature list
* Move prereq section into existing documentation section

### CODE REVIEW

**Review by borkmann — APPROVED** [review]

File: Documentation/admin.rst [hunk 1]
```diff
@@ -17,25 +17,54 @@ for a simple way to experiment on your laptop, we highly recommend using our Vag
...
-general purpose distro that runs a 4.8+ kernel by default.  However, many other popular distros have the
-ability to optionally install a kernel with version 4.8 or newer.
+General purpose Linux distros are less likely to come with a 4.8+ kernel by
+default.  As of March 2017, `Ubuntu 16.10
+<https://wiki.ubuntu.com/YakketyYak/ReleaseNotes#Linux_kernel_4.8>`_ is the
+most widely used general purpose distro that runs a 4.8+ kernel by default.
+However, many other popular distros have the ability to optionally install a
+kernel with version 4.8 or newer.
+
+Cilium will make use of later kernel versions if available. It will probe for
```
Comment (aanm) [inline Documentation/admin.rst]: It will probe **at run time** for the...?

File: Documentation/admin.rst [hunk 2]
```diff
@@ -17,25 +17,54 @@ for a simple way to experiment on your laptop, we highly recommend using our Vag
...
+^^^^^^^^^^^^^^^^^^^^
 
-Since Cilium builds on the foundation of BPF, the kernel version running on the Linux node that runs
-containers and Cilium must support a modern version of BPF.  For practical purposes, this means that
-the kernel must be 4.8.0 or newer.
+Since Cilium builds on the foundation of BPF, the kernel version running on the
+Linux node that runs containers and Cilium must support a modern version of
+BPF.  For practical purposes, this means that the kernel must be 4.8.0 or
+newer. However, we recommend to run the 4.9.x kernel releases which is a
+supported stable release of the Linux kernel.
```
Comment (aalemayhu) [inline Documentation/admin.rst]: Why not recommend the latest stable version (4.10.12)?

**Review by aalemayhu — APPROVED** [review]
Looks good to me.

File: Documentation/admin.rst (hunk 2 above)
Comment (tgraf) [reply Documentation/admin.rst]: I rephrased to "at least". This is just the minimal kernel version we recommend

File: Documentation/admin.rst (hunk 1 above)
Comment (tgraf) [reply Documentation/admin.rst]: Rephrased

**Pull request merged**